File: apps/api/app/schemas/duerp.py
from uuid import UUID

from pydantic import BaseModel, Field

from app.db.models.duerp_entry import DuerpEntryStatus, DuerpSeverity
from app.schemas.common import BaseReadModel


class DuerpEntryRead(BaseReadModel):
    organization_id: UUID
    site_id: UUID | None
    site_name: str | None
    work_unit_name: str
    risk_label: str
    severity: DuerpSeverity
    prevention_action: str | None
    status: DuerpEntryStatus
    compliance_status: str
    proof_count: int


class DuerpEntryCreateRequest(BaseModel):
    site_id: UUID | None = None
    work_unit_name: str = Field(min_length=2, max_length=160)
    risk_label: str = Field(min_length=2, max_length=200)
    severity: DuerpSeverity
    prevention_action: str | None = Field(default=None, max_length=2000)


class DuerpEntryUpdateRequest(BaseModel):
    site_id: UUID | None = None
    work_unit_name: str | None = Field(default=None, min_length=2, max_length=160)
    risk_label: str | None = Field(default=None, min_length=2, max_length=200)
    severity: DuerpSeverity | None = None
    prevention_action: str | None = Field(default=None, max_length=2000)
    status: DuerpEntryStatus | None = None

File: packages/contracts/src/duerp.ts
import type { EntityId, VersionedRecord } from "./common";
import type { ComplianceStatus } from "./compliance";

export type DuerpSeverity = "low" | "medium" | "high";
export type DuerpEntryStatus = "active" | "archived";

export interface DuerpEntryRecord extends VersionedRecord {
  organization_id: EntityId;
  site_id: EntityId | null;
  site_name: string | null;
  work_unit_name: string;
  risk_label: string;
  severity: DuerpSeverity;
  prevention_action: string | null;
  status: DuerpEntryStatus;
  compliance_status: ComplianceStatus;
  proof_count: number;
}

export interface DuerpEntryCreateRequest {
  site_id?: EntityId | null;
  work_unit_name: string;
  risk_label: string;
  severity: DuerpSeverity;
  prevention_action?: string | null;
}

export interface DuerpEntryUpdateRequest {
  site_id?: EntityId | null;
  work_unit_name?: string | null;
  risk_label?: string | null;
  severity?: DuerpSeverity | null;
  prevention_action?: string | null;
  status?: DuerpEntryStatus | null;
}
